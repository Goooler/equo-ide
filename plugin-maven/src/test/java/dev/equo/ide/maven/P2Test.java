/*******************************************************************************
 * Copyright (c) 2022 EquoTech, Inc. and others.
 *
 * This program and the accompanying materials
 * are made available under the terms of the Eclipse Public License 2.0
 * which accompanies this distribution, and is available at
 * https://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Contributors:
 *     EquoTech, Inc. - initial API and implementation
 *******************************************************************************/
package dev.equo.ide.maven;

import au.com.origin.snapshots.Expect;
import au.com.origin.snapshots.junit5.SnapshotExtension;
import java.io.IOException;
import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.extension.ExtendWith;

@ExtendWith({SnapshotExtension.class})
public class P2Test extends MavenHarness {
	@Test
	public void help(Expect expect) throws IOException, InterruptedException {
		setPom("");
		mvnw("help:describe -Dcmd=equo-ide:p2 -Ddetail")
				.snapshotBetween("Mojo: 'equo-ide:p2'", "[INFO] BUILD SUCCESS", expect);
	}
}
