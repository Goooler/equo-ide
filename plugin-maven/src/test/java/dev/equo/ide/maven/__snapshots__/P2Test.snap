dev.equo.ide.maven.P2Test.help=[
equo-ide:p2
  Description: (no description available)
  Implementation: dev.equo.ide.maven.P2Mojo
  Language: java

  Available parameters:

    all
      User property: all
      (no description available)

    detail
      User property: detail
      (no description available)

    filters
      (no description available)

    format (Default: ASCII)
      User property: format
      (no description available)

    installed (Default: false)
      User property: installed
      (no description available)

    installs
      (no description available)

    optional (Default: false)
      User property: optional
      (no description available)

    p2repos
      (no description available)

    problems (Default: false)
      User property: problems
      (no description available)

    raw
      User property: raw
      (no description available)
.
.
[INFO] ------------------------------------------------------------------------
]